proxy/envoy/ingress.go

```diff
@@ -111,8 +130,38 @@ func generateIngress(conf *IngressConfig) *Config {
 		rulesByHost[host] = append(rulesByHost[host], rule)
 	}
 
-	// Phase 2: create a VirtualHost for each host
+	// figure out which hosts are configured to terminate TLS
+	// ensure that only one TLS config is used
+	// TODO: extensive tests
+	tlsValid := true
+	tls, err := conf.Secrets.GetSecret("*")
+	if err != nil {
+		glog.Warningf("Error retrieving TLS context for wildcard: %v", err)
+		tlsValid = false
+	}
+	wildcardTLS := tls != nil
+
+	tlsHosts := make(map[string]bool)
+	for host := range rulesByHost {
+		if t, err := conf.Secrets.GetSecret(host); err != nil {
+			glog.Warningf("Error retrieving TLS context for %q: %v", host, err)
+			tlsHosts[host] = true
+		} else if t != nil {
+			if tls == nil {
+				tls = t
+			} else if string(tls.PrivateKey) != string(t.PrivateKey) || string(tls.Certificate) != string(t.Certificate) {
+				glog.Warningf("Invalid ingress configuration for %q: multiple TLS configs", host)
```

Comment:
nit: this is not an invalid configuration, but rather an unsupported one due to lack of SNI in Envoy.